### PR TITLE
Hooking up Travis with this repository.

### PR DESCRIPTION
**Pull request by fmilthaler**
This branch aims at getting this repository ready for using Travis for automated testing.

The related "opesci/data" repository is cloned alongside "opesci/inversion" as some data files will be used for testing purposes. Since some bigger data files are stored via git-lfs in "opesci/data", git-lfs is installed.

The first test will follow in the branch "binreader". Hence, this branch needs to go in before "binreader" does.

### CODE REVIEW

**Comment by fmilthaler** [comment]
Please review, comment and merge.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  - OLDDIR=${PWD}
+  - cd ..
+  - git clone https://github.com/opesci/data.git
+  - cd $OLDDIR
+
+before_script:
+  - export PYTHONPATH=`pwd`:$PYTHONPATH
+
+script:
+  - echo "run tests once merged."
```
Comment (navjotk) [inline .travis.yml]: So this is a skeleton for how the tests would be run? Seems fine to me.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+before_install:
+  - curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | sudo bash
+
+install:
+  - sudo apt-get update
+  - sudo apt-get install git-lfs
+  - git lfs install
+  - OLDDIR=${PWD}
+  - cd ..
+  - git clone https://github.com/opesci/data.git
```
Comment (mlange05) [inline .travis.yml]: We could just make the data repo a git submodule and do the clone with `--recursive`. This way we don't have to do all the directory shuffling.

**Comment by fmilthaler** [comment]
@mlange05: That is a good idea, however since the submodules are fetched at the beginning of the procedure. That is before git-lfs is installed. Hence, the data files stored with git-lfs are not downloaded. Any ideas?

**Comment by fmilthaler** [comment]
Ok, just found sth that might work: `git lfs fetch` that should update the large data files.

**Comment by mlange05** [comment]
Uh, I just realised you are setting `sudo: required`. You need to switch to the new containerised way to install apt packages like `git-lfs`. For a template check the travis setup in opesci-fd.

**Comment by fmilthaler** [comment]
Done, it is not very pretty, but it works. It looks like everything is in place now. Have a look.

**Comment by mlange05** [comment]
Oh ok, I didn't realise that this was quite so painful. But yes, it looks good now.

**Comment by mlange05** [comment]
OK, apparently I merged this too quickly. While this ran clean on the branch tests, we're now getting the following error on master:
```
$ git-lfs pull
Git LFS: (0 of 3 files) 0 B / 51.49 MB                                         
This repository is over its data quota. Purchase more data packs to restore access.
Docs: https://help.github.com/articles/purchasing-additional-storage-and-bandwidth-for-an-organization/
This repository is over its data quota. Purchase more data packs to restore access.
Docs: https://help.github.com/articles/purchasing-additional-storage-and-bandwidth-for-an-organization/
```

I'm backing this out now and re-opening the PR for further discussion.

**Comment by fmilthaler** [comment]
Oh dear, I guess we need to look for an alternative to git lfs. Didn't think we' hit the quota that quickly. I'll do some research on what else we could use.

**Pull request merged**